src/main/java/com/algorand/algosdk/crypto/Address.java

```diff
@@ -110,6 +128,38 @@ public String encodeAsString() throws NoSuchAlgorithmException {
         return res;
     }
 
+    /**
+     * verifyBytes verifies that the signature for the message is valid for the public key
+     * @param message the message that was signed
+     * @param signature
+     * @return boolean; true if the signature is valid
+     */
+    public boolean verifyBytes(byte[] message, byte[] signature) throws NoSuchAlgorithmException, InvalidKeySpecException, InvalidKeyException, SignatureException {
+        CryptoProvider.setupIfNeeded();
+        X509EncodedKeySpec pkS;
+        try {
+            // Wrap both keys in ASN.1 format.
+            SubjectPublicKeyInfo publicKeyInfo = new SubjectPublicKeyInfo(new AlgorithmIdentifier(EdECObjectIdentifiers.id_Ed25519), this.bytes);
+            pkS = new X509EncodedKeySpec(publicKeyInfo.getEncoded());
+        } catch (IOException e) {
+            throw new RuntimeException("could not parse raw key bytes", e);
+        }
+        // prepend the message prefix
+        byte[] prefixBytes = new byte[message.length + BYTES_SIGN_PREFIX.length];
+        System.arraycopy(BYTES_SIGN_PREFIX, 0, prefixBytes, 0, BYTES_SIGN_PREFIX.length);
+        System.arraycopy(message, 0, prefixBytes, BYTES_SIGN_PREFIX.length, message.length);
+        
+        // create public key
+        KeyFactory kf = KeyFactory.getInstance(KEY_ALGO);
+        PublicKey pk = kf.generatePublic(pkS);
+        
+        // verify signature
+        java.security.Signature sig = java.security.Signature.getInstance(SIGN_ALGO);
+        sig.initVerify(pk);
+        sig.update(prefixBytes);
+        return sig.verify(signature);
+    }
```

Comment:
Can you add a unit test? (Unless there's a cucumber test already that tests at this level of granularity?)

Reply:
There's a positive and negative test in testSignBytes for verifying.